src/libYARP_manager/src/manager.cpp

```diff
@@ -370,6 +391,22 @@ Broker* Manager::createBroker(Module* module)
     return (new ScriptYarprunBroker(module->getBroker()));
 }
 
+bool Manager::removeBroker(Executable* exe)
+{
+    if (exe == nullptr)
+    {
+        return false;
+    }
+    else if(exe->state() == RUNNING)
```

Comment:
It is not better to delegate this (i.e. stopping the module) to the Manager class and the user?

Reply:
Hi @apaikan ! This is an extra check to be 100% sure, but I think that `exe->state() == RUNNING` is never true because the broker can be switched only changing the entry of "host" and it is editable only when the module is **not** running.
It is actually delegated to `yarp::manager::Manager` class 🤔